src/classes/STG_PanelRD_CTRL.cls

```diff
@@ -271,4 +276,97 @@ public with sharing class STG_PanelRD_CTRL extends STG_Panel {
 
         return '';
     }
+
+    /***
+    * @description The current RD naming format being stored on the RD Setting Name Format
+    */
+    public String displayedRDNamingFormat {
+        get {
+            displayedRDNamingFormat = convertToRDNamingLabel(getCurrentRDNamingSetting());
+            return displayedRDNamingFormat;
+        }
+        set;
+    }
+
+    /**
+    * @description currently selected RD naming format 
+    */
+    public String selectedRDNamingFormat {get;set;}
+
+    /***
+    * @description Return a list of picklist value to be selected for RD Auto naming format
+    */
+    public List<SelectOption> rdNamingOptions {
+        get {
+            if (rdNamingOptions == null) {
+                rdNamingOptions = getRdNamingOptions();
+            }
+            return rdNamingOptions;
+        }
+        set;
+    }
+
+    /**
+    * @description Evaluate and assign the current RD Naming format setting api
+    * @return String
+    */
+    private String getCurrentRDNamingSetting () {
+        String currentFormat;
+        if (!STG_Panel.stgService.stgRD.EnableAutomaticNaming__c) {
+            currentFormat = '';
+
+        } else if (String.isBlank(STG_Panel.stgService.stgRD.RecurringDonationNameFormat__c)) {
+            currentFormat = RD2_NamingService.RD_NAMING_OPTION_1;
+
+        } else {
+            currentFormat = STG_Panel.stgService.stgRD.RecurringDonationNameFormat__c;
+        }
+
+        selectedRDNamingFormat = currentFormat;
+
+        return currentFormat;
+    }
+
+    /***
+    * @description Generate a list of RD naming format options for selection
+    */
+    private List<SelectOption> getRdNamingOptions() {
```

Comment:
This method and convertToRDNamingLabel are nearly identical. You can probably just call convertToRDNamingLabel() from this method for each of the naming options.

Also, in my branch I changed from the English words (Recurring, Recurring Donation) to tokens - $RECURRING_NAME_SUFFIX$ and $RECURRING_OBJECT_NAME$ respectively just to avoid using english words that could get changed by a customer. Plus I think it could allow us to expand usage at a future point.